File: Research/LangComparePrime/Prime.hs
import Data.List

divisibleRec :: Int -> Int -> Bool
divisibleRec i j 
  | j == 1         = False 
  | i `mod` j == 0 = True 
  | otherwise      = divisibleRec i (j-1)

divisible :: Int -> Bool
divisible i = divisibleRec i (i-1)

--r = [ x | x <- [2..20000], not (divisible x)]

main :: IO ()
main = print $ foldl' (\s x -> if divisible x then s else s+1) 0 [2..20000]
--main = print (length r)

File: Research/LangComparePrime/prime.c
#include <stdio.h>

int divisibleRec(int i, int j){
  while (j>0) {
  if(j==1){ return 0; }
  else if(i%j==0){ return 1; }
  else{ //return divisibleRec(i,j-1); 
    j = j - 1;
  }
  }
}

int divisible(int i){ return divisibleRec(i, i-1); }

int main(void){
  int i, count =0;
  for(i=2; i<20000; ++i){
    if(!divisible(i)){
      count = count+1;
    }
  }
  printf("number of primes = %d\n",count);
  return 0;
}
